decimal-macros/src/lib.rs

```diff
@@ -17,11 +16,14 @@ use syntax::ast::{ExprKind, TokenTree, LitKind, StrStyle};
 use decimal::d128;
 
 fn d128_lit<'cx>(cx: &'cx mut ExtCtxt, sp: Span, tts: &[TokenTree]) -> Box<MacResult + 'cx> {
+    // Turn input into a string literal
+    // e.g. d128!(0.1) -> "0.1", d128!(NaN) -> "NaN"
```

Comment:
Aha so was I before this - [this](https://doc.rust-lang.org/book/compiler-plugins.html) and [this](https://www.gulshansingh.com/posts/how-to-write-a-rust-syntax-extension/) really helped. Also the [docs](https://manishearth.github.io/rust-internals-docs/syntax/index.html).